eng/packages/http-client-csharp/generator/TestProjects/Local/Basic-TypeSpec/src/Generated/BasicTypeSpecClient.cs

```diff
@@ -68,12 +68,12 @@ public BasicTypeSpecClient(Uri endpoint, ApiKeyCredential keyCredential, BasicTy
         /// </summary>
         /// <param name="headParameter"></param>
         /// <param name="queryParameter"></param>
-        /// <param name="optionalQuery"></param>
         /// <param name="context"> The request options, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        /// <param name="optionalQuery"></param>
         /// <exception cref="ArgumentNullException"> <paramref name="headParameter"/> or <paramref name="queryParameter"/> is null. </exception>
         /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
         /// <returns> The response returned from the service. </returns>
-        public virtual Response SayHi(string headParameter, string queryParameter, string optionalQuery, RequestContext context)
+        public virtual Response SayHi(string headParameter, string queryParameter, RequestContext context, string optionalQuery = null)
```

Comment:
It looks wrong to me. Do we have any new design that RequestContext could not be the last parameter?

Reply:
To make RequestContext required, all the parameters should change to required.

Reply:
yeah this is incorrect.
Does this case break in MGC as well? If so we should fix it there.

Reply:
This change comes from MGC version bump:
https://github.com/microsoft/typespec/blob/82995bf75d6a192dda1ace66fff80e02308cc62d/packages/http-client-csharp/generator/TestProjects/Local/Unbranded-TypeSpec/src/Generated/UnbrandedTypeSpecClient.cs#L73

Reply:
Then MGC should be fixed as well.

Reply:
RequestContext is required here because making it optional would cause ambiguous callsites due to the convenience method having the same first two parameters. The rule is that RequestContext should be optional as long as making it optional would not result in ambiguous callsites.

Reply:
But I agree it looks odd that RequestContext is not the last parameter. I will follow up on this with an issue and confirmation of what the ideal behavior should be.

Reply:
Filed https://github.com/microsoft/typespec/issues/5065